api/test-context/src/test_context.rs

```diff
@@ -325,16 +325,127 @@ impl TestContext {
         )
     }
 
+    pub async fn create_and_fund_account(&mut self) -> LocalAccount {
+        let account = self.gen_account();
+        self.commit_block(&vec![self.mint_user_account(&account)])
+            .await;
+        account
+    }
```

Comment:
the problem with this code is that root account is regenerated each time and the LocalAccount doesn't read the chain to determine the correct sequence number :facepalm: 

I intentionally just wrote my code independently of this file and said... we need to at some point in time come back in here with a heavy opinion and fix it.

Reply:
I updated test_context to refresh the root account's sequence number every time in root_account(). Not the cleanest solution as that makes that function async + requires await everywhere, but it works for now. We can go back and refactor test_context + api tests later.